packages/api/src/beacon/server/events.ts

```diff
@@ -38,7 +38,9 @@ export function getRoutes(config: ChainForkConfig, api: ServerApi<Api>): ServerR
             void api.eventstream(req.query.topics, controller.signal, (event) => {
               try {
                 // If the request is already aborted, we don't need to send any more events.
-                if (req.raw.destroyed) return;
+                // This prevents "Cannot set headers after they are sent to the client" errors
+                // during shutdown or when client aborts abruptly.
+                if (res.raw.writable) return;
```

Comment:
I believe you want here if response is not writeable. 

```suggestion
                if (!res.raw.writable) return;
```

Reply:
If req is aborted then response will not be writeable, both are strictly related. Not sure why the earlier is not working. Seems like a bug in fastify implementation.

Reply:
yeah, you are right. Was looking so many different things, completely missed that. This fixed this issue but obviously broke the eventstream api.